### PR TITLE
Clean up test classes

### PR DESCRIPTION
**Pull request by jerith**
We're constructing test classes from mixins in a bunch of places when we should just be using some preconstructed base classes. (This work is factored out of some stuff I started to do in #246.)

### CODE REVIEW

File: go/api/go_api/tests/test_session.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+from go.vumitools.tests.utils import GoTestCase
 
 
-class SessionStoreTestCase(TestCase, GoPersistenceMixin):
+class SessionStoreTestCase(GoTestCase):
     def mk_session_store(self, session_key=None):
         ss = SessionStore(session_key)
+        # We don't grab the Riak manager here because we don't use Riak in
+        # these tests.
+        self._persist_redis_managers.append(ss.vumi_api.redis)
```
Comment (hodgestar) [inline go/api/go_api/tests/test_session.py:12]: Why do we have to manually grab the `redis` store here? And anyway to avoid it?

**Comment by hodgestar** [comment]
Other than the one question, looks good.

File: go/api/go_api/tests/test_session.py (hunk 1 above)
Comment (jerith) [inline go/api/go_api/tests/test_session.py:12]: We have to do that everywhere we create a VumiApi object, it's just usually hidden inside test helpers.

Comment (hodgestar) [inline go/api/go_api/tests/test_session.py:12]: Cool. Would be nice to hide all of this somehow but let's worry about that some other time.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**